testscript/testscript.go

```diff
@@ -316,6 +316,11 @@ func (ts *TestScript) setup() string {
 			"exe=",
 		)
 	}
+	// If we are collecting coverage profiles for merging into the main one,
+	// ensure the environment variable is forwarded to sub-processes.
+	if coverdir := os.Getenv("TESTSCRIPT_COVER_DIR"); coverdir != "" {
```

Comment:
... or just put it in the `Vars` initializer above (it doesn't matter if it's empty).